src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPage.cs

```diff
@@ -37,6 +40,11 @@ public override IFileOriginsPageViewModel CreateViewModel(FileOriginsPageContext
             ServiceProvider.GetRequiredService<IArchiveInstaller>(),
             ServiceProvider.GetRequiredService<IRepository<DownloadAnalysis.Model>>(),
             ServiceProvider.GetRequiredService<IConnection>(),
+            ServiceProvider.GetRequiredService<IFileSystem>(),
+            ServiceProvider.GetRequiredService<ILogger<FileOriginsPageViewModel>>(),
+            ServiceProvider,
+            ServiceProvider.GetRequiredService<IFileOriginRegistry>(),
+            ServiceProvider.GetRequiredService<IOSInterop>(),
             ServiceProvider.GetRequiredService<IWindowManager>()
```

Comment:
This is getting ridiculous. Just pass the `LoadoutId` and the `IServiceProvider` and let the constructor fetch the other stuff.